extensions-core/orc-extensions/src/test/java/org/apache/druid/data/input/orc/OrcHadoopInputRowParserTest.java

```diff
@@ -217,54 +217,70 @@ private static HadoopDruidIndexerConfig loadHadoopDruidIndexerConfig(String conf
     return HadoopDruidIndexerConfig.fromFile(new File(configPath));
   }
 
-  private static OrcStruct getFirstRow(Job job, String orcPath) throws IOException, InterruptedException
+  private static OrcStruct getFirstRow(Job job, String orcPath) throws IOException
   {
     File testFile = new File(orcPath);
     Path path = new Path(testFile.getAbsoluteFile().toURI());
-    FileSplit split = new FileSplit(path, 0, testFile.length(), null);
+    FileSplit split = new FileSplit(path, 0, testFile.length(), new String[]{"host"});
 
-    InputFormat inputFormat = ReflectionUtils.newInstance(
+    InputFormat<NullWritable, OrcStruct> inputFormat = ReflectionUtils.newInstance(
         OrcInputFormat.class,
         job.getConfiguration()
     );
-    TaskAttemptContext context = new TaskAttemptContextImpl(job.getConfiguration(), new TaskAttemptID());
-
-    try (RecordReader reader = inputFormat.createRecordReader(split, context)) {
-
-      reader.initialize(split, context);
-      reader.nextKeyValue();
-      return (OrcStruct) reader.getCurrentValue();
+    RecordReader<NullWritable, OrcStruct> reader = inputFormat.getRecordReader(
+        split,
+        new JobConf(job.getConfiguration()),
+        null
+    );
+    try {
+      final NullWritable key = reader.createKey();
+      final OrcStruct value = reader.createValue();
+      if (reader.next(key, value)) {
+        return value;
+      } else {
+        throw new NoSuchElementException();
+      }
+    }
+    finally {
+      reader.close();
     }
   }
 
-  private static List<InputRow> getAllRows(HadoopDruidIndexerConfig config)
-      throws IOException, InterruptedException
+  // This method returns a list of the final rows. This is definitely a bug, but it should be better to delete
```

Comment:
thanks for fixing this, I noticed a while ago but then forgot about it :rofl:

Reply:
Oh I actually fixed this method but forgot to remove the comment. Will remove it.